rafiki/model/model.py

```diff
@@ -169,8 +171,12 @@ def test_model_class(model_file_path, model_class, task, dependencies, \
         _check_model_inst(model_inst)
 
         _print_header('Checking training & evaluation of model...')
-        model_inst.train(train_dataset_uri)
-        score = model_inst.evaluate(test_dataset_uri)
+        if task == TaskType.TABULAR_REGRESSION or TaskType.TABULAR_CLASSIFICATION:
```

Comment:
This won't work on Rafiki because it breaks the contract given by [Rafiki's base model API](https://nginyc.github.io/rafiki/docs/latest/src/python/rafiki.model.html) for the `train` and `evaluate` method. Currently, we aim to treat all Rafiki models abstractly with the same interface. 

I believe the addition of `features` & `target` arguments to be passed to the model instance is to customize which columns in a dataset are used as features & which column is to be predicted. To conform to Rafiki's base model class & model framework, the dataset format for `TABULAR_REGRESSION` and `TABULAR_CLASSIFICATION` task should be defined in a way that *the dataset file completely & unambiguously defines the ML task to be performed on the dataset*.  On the top of my head, I can think of 2 ways to do this:

1. Define the dataset format such that the first (N-1) columns are features, last column is the target. Then, write a dataset pre-processing/loading script to alter the original `.csv` file
2. Define the dataset format such that it is a `.zip` file containing the original `.csv` file and an additional metadata file e.g. `meta.json` that describes which columns to use as features, which column is the target

Reply:
A third option is to add **kwargs to the train and predict function and let the model developer to  add the documents on the allowed arguments and parse those arguments in the train/predict functions.

Reply:
I choose the second way and the first way will be applied if the `.json` file is empty